x-pack/plugin/esql/qa/testFixtures/src/main/resources/spatial.csv-spec

```diff
@@ -3024,3 +3027,74 @@ wkt:keyword                |pt:cartesian_point
 "POINT(111)"               |null
 // end::to_cartesianpoint-str-parse-error-result[]
 ;
+
+cityInCityBoundaryOnNulls
+# this test ensures that the non-grouping ST_EXTENT_AGG is able to gracefully handle missing cartesian shapes
+
+required_capability: st_intersects
+required_capability: st_extent_agg_null_support
+
+FROM airport_city_boundaries
+| WHERE city_boundary IS NULL
+| EVAL in_city = ST_INTERSECTS(city_location, city_boundary)
+| STATS count=COUNT(*) BY in_city
+| SORT count ASC
+| EVAL cardinality = CASE(count < 10, "very few", count < 100, "few", "many")
+| KEEP cardinality, in_city
+;
+
+cardinality:k | in_city:boolean
+very few      | null
+;
+
+stExtentCartesianShapesGroupingOnNulls
+# this test ensures that the non-grouping ST_EXTENT_AGG is able to gracefully handle missing cartesian shapes
+
+required_capability: st_extent_agg
+required_capability: st_extent_agg_null_support
+
+FROM cartesian_multipolygons
+| WHERE shape IS NULL
+| EVAL key = SUBSTRING(name,1,3)
+| STATS extent = ST_EXTENT_AGG(shape), count = COUNT() BY key
+| KEEP count, key, extent
+| SORT count DESC, key ASC
+;
+
+count:long | key:keyword | extent:cartesian_shape
+1          | Mis         | null
+;
+
+stExtentCartesianShapesGroupingNoDocValuesOnNulls
+# this test ensures that the non-grouping ST_EXTENT_AGG is able to gracefully handle missing cartesian shapes
+
+required_capability: st_extent_agg
+required_capability: st_extent_agg_null_support
+
+FROM cartesian_multipolygons_no_doc_values
+| WHERE shape IS NULL
+| EVAL key = SUBSTRING(name,1,3)
+| STATS extent = ST_EXTENT_AGG(shape), count = COUNT() BY key
+| KEEP count, key, extent
+| SORT count DESC, key ASC
+;
+
+count:long | key:keyword | extent:cartesian_shape
+1          | Mis         | null
+;
+
+stExtentCartesianShapesNonGroupingOnNulls
```

Comment:
I noticed all tests are on cartesian data, never geo data. This could be sufficient, since the changes are identical across the two types, but ideally tests on geo would also be nice.